src/Database/Query.php

```diff
@@ -1393,6 +1394,27 @@ public function into($table)
     }
 
     /**
+     * Creates an expression that refers to an identifier. Identifiers are used to refer to field names and allow
+     * the SQL compiler to apply quotes or escape the identifier.
+     *
+     * The value is used as is, and you might be required to use aliases or include the table reference in
+     * the identifier. Do not use this method to inject SQL methods or logical statements.
+     *
+     * ### Example
+     *
+     * ```
+     * $query->newExp()->lte('count', $query->identifier('total'));
+     * ```
+     *
+     * @param string $identifier The identifier for an expression
+     * @return ExpressionInterface
+     */
+    public function identifier($identifier)
+    {
+        return new IdentifierExpression($identifier);
```

Comment:
Could marking this as to return `ExpressionInterface` be considered a little too lax maybe?

From a strictly API context focused point of view, it would mean that we cannot expect the method to return a `IdentifierExpression` compatible implementation (even if the possibility for the core to ever actually return an incompatible object is rather low), and thus have to assume that the returned object is possibly unsafe to be passed into parts of code where an identifier expression interface is explicitly required / expected / tested for.